joomla/joomla-cms
# Update plugins/fields/media/media.php to prevent deprecated message caused if $value empty.

### Summary of Changes
PHP error was encountered when test instructions below are met:

`json_decode(): Passing null to parameter #1 ($json) of type string is deprecated`

If $value is null at line 86 in plugins/fields/media/media.php, then the deprecated error occurs.

Added if statement to checkValue function to check $value is set before trying to decode the string.


### Testing Instructions

- Create a custom field for articles that is a Media type, that's also set to optional (ie Required = No)
- Create an article, leaving the optional custom fields blank.
- View the article in the front end.
- Turn on Error Reporting for the site to Development so you can see the messages on screen.

(ShackOpenGraph also installs Social Images fields that are Media type and trigger the error)

### Actual result BEFORE applying this Pull Request

If there's no image in the custom field, then the code in the plugin does a check on the $value provided to the checkValue function. As there's no value, json_decode triggers 
`json_decode(): Passing null to parameter #1 ($json) of type string is deprecated`

### Expected result AFTER applying this Pull Request
The checkValue function will now firstly check whether $value is null, and then only proceed with the rest of the function when $value has a value provided to it.

This in turn stops the deprecated message being generated.

### Link to documentations
Please select:
